addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -248,11 +304,11 @@ export function closestBlock(node) {
  * Returns the deepest child in last position.
  *
  * @param {Node} node
- * @param {boolean} [stopAtBlock=false]
+ * @param {Function} filterOutFunction
```

Comment:
If I'm not mistaken, everywhere else we use a positive predicate (typically, `Array.filter` to say what to keep). Couldn't we do the same here? You'd have `inline && node => !isBlock(node)` instead of `inline && isBlock` which I find semantically less confusing.
Besides it's not really a `filter` given that this only returns one node. Rather, it will stop once the node satisfies the predicate.
So `lastLeaf(node, isBlock)` returns the first `lastChild` that is not a block.
Isn't `lastLeaf(node, predicate)` -> `lastLeaf(node, node => !isBlock(node))` clearer for that? Still weird behavior but if that's in fact what we need...

Btw docstring: optional arguments are between square braces.

Reply:
I have tried to change it but it is confusing in another way.
Can we do that in another potential refactor ?
I have changed the name of the function to be `stopTraverseFunction` and added `[stopTraverseFunction]` in the comments.

Reply:
That's fine by me. It's already better with `stopTraverseFunction` but let's not forget the refactor for _later_ ^^